pkg/apis/pipeline/v1beta1/taskrun_validation_test.go

```diff
@@ -64,6 +64,110 @@ func TestTaskRun_Validate(t *testing.T) {
 			ObjectMeta: metav1.ObjectMeta{Name: "taskrname"},
 		},
 		wc: apis.WithinDelete,
+	}, {
+		name: "propagating params with taskrun",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ArrayOrString{
+						ArrayVal: []string{"hello", "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Steps: []v1beta1.Step{{
+						Name:    "echo",
+						Image:   "ubuntu",
+						Command: []string{"echo"},
+						Args:    []string{"$(params.task-words[*])"},
+					}},
+				},
+			},
+		},
+		wc: config.EnableAlphaAPIFields,
+	}, {
+		name: "propagating params with taskrun different names",
+		taskRun: &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: "tr"},
+			Spec: v1beta1.TaskRunSpec{
+				Params: []v1beta1.Param{{
+					Name: "task-words",
+					Value: v1beta1.ArrayOrString{
+						ArrayVal: []string{"hello", "task run"},
+					},
+				}},
+				TaskSpec: &v1beta1.TaskSpec{
+					Params: []v1beta1.ParamSpec{{
+						Name: "default-task-words",
```

Comment:
this test case is a bit confusing-- I'm understanding this as the task declares a totally separate param that happens to not be used, but the name of the separate param is "default" + the name of the param that is used. What's this trying to test?

Reply:
I think what's confusing is specifically referring to params defined inline as "default" params. (as you know) params can have defaults, which is separate from the concept of a param being a "default param". The idea of param defaults only really makes sense in a task ref (rather than an inline task spec) IMO. I think this test case would be less confusing if you just removed "default" from all the variable names, and changed "default-task-words" and "default-echo" to different names. However if you disagree that's fine.

Reply:
Ah ok. Sure I can remove `default` from there. I will post here once I update the code.

Reply:
Done.